zookeeper-server/src/main/java/org/apache/zookeeper/server/ZooKeeperServer.java

```diff
@@ -2052,4 +2052,16 @@ private boolean buffer2Record(ByteBuffer request, Record record) {
         return rv;
     }
 
+    /** gets maximum number of connections in ZooKeeper */
+    public int getMaxCnxns() {
+        if (serverCnxnFactory != null) {
+            return serverCnxnFactory.getMaxCnxns();
+        }
+        if (secureServerCnxnFactory != null) {
+            return secureServerCnxnFactory.getMaxCnxns();
+        }
```

Comment:
A peer can operate both a `serverCnxnFactory` and a `secureServerCnxnFactory` simultaneously. Your accounting is done on each separately so a peer is allowed to host up to `2*maxCnxns` provided they are of appropriate types. If this is intentional then it should be mentioned in the documentation ("maxCnxns per client port").

Reply:
Done, thanks